pkg/option/config.go

```diff
@@ -3164,8 +3165,27 @@ func sanitizeIntParam(paramName string, paramDefault int) int {
 	return intParam
 }
 
+// validateConfigmap checks whether the flag exists and validate the value of flag
+func validateConfigmap(cmd *cobra.Command, m map[string]interface{}) (error, string) {
+	// validate the config-map
+	for key, value := range m {
+		if val := fmt.Sprintf("%v", value); val != "" {
+			flags := cmd.Flags()
+			// check whether the flag exists
+			if flag := flags.Lookup(key); flag != nil {
+				// validate the value of flag
+				if err := flag.Value.Set(val); err != nil {
```

Comment:
Is this going to overwrite values that might be set in the command line? This seems unexpected to me.

Reply:
> Is this going to overwrite values that might be set in the command line? This seems unexpected to me.

No, it isn't. I checked it. For example, in case of bool type of value will be done this function: https://github.com/spf13/pflag/blob/master/bool.go#L20
It doesn't overwrite values from CLI. Also the function "validateConfigmap"  is executed only for ConfigMap